data-plane/core/src/test/java/dev/knative/eventing/kafka/broker/core/OrderedAsyncExecutorTest.java

```diff
@@ -56,26 +58,26 @@ private static Stream<Arguments> inputArgs() {
 
   @ParameterizedTest(name = "with delay {0}ms and tasks {1}")
   @MethodSource("inputArgs")
-  public void shouldExecuteInOrder(final long millis, final int tasks, Vertx vertx) throws InterruptedException {
+  public void shouldExecuteInOrder(final long millis, final int tasks, final Vertx parentVertx) throws InterruptedException {
     Random random = new Random();
 
     // Deploy the verticle
     AVerticle verticle = new AVerticle();
     CountDownLatch startVerticleLatch = new CountDownLatch(1);
-    vertx.deployVerticle(verticle, v -> startVerticleLatch.countDown());
+    parentVertx.deployVerticle(verticle, v -> startVerticleLatch.countDown());
     startVerticleLatch.await();
 
     // Rewrite the vertx instance in order to make sure we run always in the same context
-    vertx = verticle.getVertx();
+    final var vertx = verticle.getVertx();
 
     CountDownLatch tasksLatch = new CountDownLatch(tasks);
     List<Integer> executed = new ArrayList<>(tasks);
 
     OrderedAsyncExecutor asyncExecutor = new OrderedAsyncExecutor();
 
     for (int i = 0; i < tasks; i++) {
-      Supplier<Future<?>> task = generateTask(vertx, random, millis, i, tasksLatch, executed);
-      vertx.runOnContext((v) -> asyncExecutor.offer(task));
+      final var n = i;
```

Comment:
why this explicit declaration?

Reply:
lambda captured variables have to `final`.